_posts/2020-09-19-github-actions-with-faasd.md

```diff
@@ -0,0 +1,218 @@
+
+
+## Prerequisites
+
+* A VM exposed to the Internet
+
+For this tutorial we'll be using a [DigitalOcean](https://digitalocean.com/) Droplet.
+
+* faas-cli
+
+faas-cli is a command line tool used to interact with OpenFaaS, from deploying your functions to viewing logs.
+Get the installation instructions [here](https://docs.openfaas.com/cli/install/)
+
+> You can follow along with the code used in this tutorial from this [repo](https://github.com/utsavanand2/faasd-hello)
+
+## Create a Droplet with faasd
+
+Create a DigitalOcean Droplet with a Ubuntu distribution, with any plan according to what you want to run, faasd being light weight I'm going to use the $5 basic plan.
+
+![Choose a distribution and a plan](/images/2020-09-19-github-actions-faasd/choose-distro.jpg)
+
+Under `Select additional options`, select `User data` and paste in the contents of the cloud-config file from [here](https://github.com/openfaas/faasd/blob/master/cloud-config.txt)
+
+![Select additional options](/images/2020-09-19-github-actions-faasd/additional-options-cloud-config.jpg)
+
+
+![faasd droplet](/images/2020-09-19-github-actions-faasd/faasd-droplet.jpg)
+
+Within as little as 30 seconds, your faasd instance will be up and running.
+
+## Get the OpenFaaS gateway password
+
+Get the gateway password by running the following commands
+
+```sh
+export IP="143.110.184.36"
+export PASSWORD=$(ssh root@$IP "sudo cat /var/lib/faasd/secrets/basic-auth-password")
+```
+
+Assuming that you have faas-cli already installed, login to faasd with faas-cli with the following commands
+
+```sh
+export OPENFAAS_URL=http://$IP:8080
+echo $PASSWORD | faas-cli login --password-stdin -u admin -g $OPENFAAS_URL
+```
+
+## Create an OpenFaaS function
+Now that we've logged into faasd with faas-cli, we're ready to deploy functions to it.
+I'm going to deploy a golang function for this tutorial.
+```sh
+# create a new directory for our project
+mkdir faasd-hello
+# pull golang-http templates
+faas-cli template store pull golang-http
+# create a new function with faas-cli
+faas-cli new faasd-hello --lang golang-http --prefix utsavanand2
+```
+
+If you look into the root directory of the project, faas-cli has created two additional files:
+
+```
+./faasd-hello/handler.go
+./faasd-hello.yml
+```
+
+The content of our handler looks something like this:
+![./faasd-hello/handler.go](/images/2020-09-19-github-actions-faasd/handler-go.jpg)
+
+The YAML file is used to configure the deployment and the build and runtime environment of the function
+~[./faasd-hello.yml](/images/2020-09-19-github-actions-faasd/faasd-hello-yaml.jpg)
+
+## Setup a workflow file for Github Actions
+In the root directory of the project run:
+
+```sh
+mkdir -p .github/workflows
+nano ./github/workflows/main.yml
+```
+
+And paste in the following contents into main.yml file
+```yaml
+name: CI-CD
+
+on:
+  push:
+    branches: 
+      - master
+
+jobs:
+  func-build:
```

Comment:
How does this compare to what @LucasRoesler wrote?

Were you not able to use the workflow from Crazy Max that I advised you to look at?

https://github.com/crazy-max/ghaction-docker-buildx

Reply:
IMHO we should have a blog post on Multi arch builds with buildx in the next blog post.
I've looked at Lucas's post, and he is using his own GitHub Action that he has built, which has the faas-cli version hard-coded. Whereas I'm using the docker container for faas-cli, making it easier to switch versions of the cli.

Reply:
I would recommend just using the published faas-cli docker image (as it is written here), it didn't exist when i wrote my post